torch/_inductor/kernel/flex_attention.py

```diff
@@ -2539,7 +2538,14 @@ def flex_attention_backward(*args, **kwargs):
     grad_lse_exp2, delta = maybe_realize([grad_lse_exp2, delta])
 
     # # see NOTE:[TritonTemplates with multiple outputs]
-    grad_query = empty_like(query)
+    query_size = [Bq, Hq, seq_len_q, qk_head_dim]
+    grad_query_strides = infer_dense_strides(query_size, query.get_stride())
+    grad_query = empty_strided(
+        query_size,
+        stride=[sympy.sympify(s) for s in grad_query_strides],
+        dtype=query.get_dtype(),
+        device=query.get_device(),
+    )
```

Comment:
in your test, what different strides do you get for `grad_query` before/after your change?

I could believe this is right. But one thing I'm surprised by is that it looks like `empty_like()` effectively calls these same two functions under the hood when you run it with the default `MemoryFormat::Preserve (`infer_dense_strides` + `empty_strided`). See here: https://github.com/pytorch/pytorch/blob/50d4698ac8c12ad8399773aa157d25316c7c345e/aten/src/ATen/native/TensorFactories.cpp#L474